libs/debugging/CMakeLists.txt

```diff
@@ -4,17 +4,20 @@
 # Distributed under the Boost Software License, Version 1.0. (See accompanying
 # file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
-cmake_minimum_required(VERSION 3.6.3 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.3.2 FATAL_ERROR)
 
 set(debugging_headers
   hpx/debugging/demangle_helper.hpp
+  # do not add hpx/debugging/print.hpp as it includes threads
+  # instead users should add this manually on demand
```

Comment:
It would be nice if the generic parts of `print.hpp` stayed in `print.hpp` and the threading related specializations moved to a separate file outside of this module. Independent of that, the header should still go into this list. Leaving it out doesn't get rid of any dependencies.

Reply:
I agree. Unfortunately, the most useful info sometimes is the thread id stuff, so I always use it. where should we put this debug printer so that it doesn't cause unwanted dependencies. if I add print.hpp to this, then debugging.hpp pulls in threads and messes up otherwise clean dependencies. (there is some other stuff that uses debugging.hpp and that shouldn't need threads)

Reply:
Keep the non-thread related functionality in `print.hpp`, but move the customizations into `thread_print.hpp` or something like that in one of the scheduler directories (most likely `hpx/runtime/threads/detail`). A lot of this customization looks like it's also duplicating functionality that the format module does, but let's discuss that live tomorrow.

Reply:
One thing I am not sure I agree with or understand - is the rationale for adding `print.hpp` to the CMakeLists file in the first place. Adding it to the list of headers causes it to be added to the 'module header' which means that `#include <hpp/debugging.hpp>` pulls in `print.hpp`. Is it intended that if you include a module, you have to include every single header declared in that module - this seems to be a very expensive way of exposing modules.
Is it intended that users `#include` the whole module, or are they encouraged to only include the headers they are using?

Reply:
I don't think we have a clearly defined policy at this point. Both options have their advantages and disadvantages, Conceptually, all headers that are part of the module will be considered when doing the cyclic dependency analysis, regardless whether a particular header is included by the main module header file or not.

I personally weakly prefer including the main module header in modules that use a particular other module, but refer to just the files needed from inside the same module.